### ISSUE TITLE
I still get the FCM message well on Android P

### ISSUE THREAD

**Issue by HanSeungBeom** [firebase/quickstart-android#1015]
### Step 1: Describe your environment

  * Android device: Galaxy S10
  * Android OS version: Android 9.0
  * Google Play Services version: 12451000
  * Firebase/Play Services SDK version: 17.3.4
  
## Android Document Guide
![image](https://user-images.githubusercontent.com/21086676/70437280-95fa1080-1ace-11ea-9b94-dca123764475.png)

The above information is contained in the Android development document.
 for my testing, I changed the settings(`app>battery>background restriction on`), sent a push message, with device unplugged.
But the message was well received. Has this feature been applied yet?

my app targetsdkversion is 28.

#### Observed Results:

I sent push message to the test device, and received.
  
#### Expected Results:

I expected not to receive a push message because the test phone's os `9.0` and `app>battery>background restriction on`.

Is there anything I missed?

**Comment by samtstern** [comment]
@HanSeungBeom that's a good question!

@kroikie does this apply to both notification and data messages?  Also is it possible that an OEM could override this behavior?

**Comment by kroikie** [comment]
@HanSeungBeom I did checking around and that restriction is not **yet** implemented. It will be turned on in an up coming update. I don't have a timeline for this update however I would recommend that you operate as if it were already implemented to avoid unexpected behavior in the future. I have also softened the language in the docs since the FCM restriction is not yet in operation, thanks for pointing that out.
